src/main/kotlin/no/nav/familie/ef/infotrygd/service/PeriodeService.kt

```diff
@@ -5,17 +5,28 @@ import no.nav.familie.ef.infotrygd.model.StønadType.BARNETILSYN
 import no.nav.familie.ef.infotrygd.repository.PeriodeRepository
 import no.nav.familie.ef.infotrygd.rest.api.Periode
 import no.nav.familie.ef.infotrygd.rest.api.PeriodeRequest
+import no.nav.familie.ef.infotrygd.utils.InfotrygdPeriodeUtil
 import org.springframework.stereotype.Service
 
 @Service
 class PeriodeService(private val periodeRepository: PeriodeRepository) {
 
     fun hentPerioder(request: PeriodeRequest): Map<StønadType, List<Periode>> {
-        val perioder = periodeRepository.hentPerioder(request).groupBy({ it.first }) { it.second }.toMutableMap()
+        val perioder =
+            periodeRepository.hentPerioder(request).groupBy({ it.first }) { it.second }
+                .toMutableMap()
         perioder[BARNETILSYN] = hentBarnetilsynPerioderMedBarn(perioder)
         return perioder
     }
 
+    fun hentSammenslåttePerioder(request: PeriodeRequest): Map<StønadType, List<Periode>> {
+        val perioder =
+            periodeRepository.hentPerioder(request).groupBy({ it.first }) { it.second }.mapValues { slåSammenPerioder(it.value) }
+                .toMutableMap()
+        perioder[BARNETILSYN] = slåSammenPerioder(hentBarnetilsynPerioderMedBarn(perioder))
+        return perioder
```

Comment:
Denne kan gjennbruke `hentPerioder`,
```suggestion
        return hentPerioder(request).map { it.key to slåSammenPerioder(it.value) }.toMap()
```